src/internal/bind-this-to-member-functions.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Bind all functions of the given instance to itself so you can use them independently.
+ *
+ * @internal
+ *
+ * @param instance The class instance of which the methods are to be bound to itself.
+ *
+ * @example
+ * const someModule = new SomeModule(faker);
+ * bindThisToMemberFunctions(someModule); // Usually called inside the constructor passing `this`
+ * const someMethod = someModule.someMethod;
+ * someMethod(); // Works
+ */
+export function bindThisToMemberFunctions<TClass extends { new (): any }>(
```

Comment:
IMO this should be the other way round:

```suggestion
export function bindMemberFunctionsToThis<TClass extends { new (): any }>(
```

Reply:
But the `this` scope already has the member function :thinking:

Reply:
But this gets bound to the member function, not the function to this

Reply:
I'm not sure whether I understand you here.

---

Maybe `bindThisOnMemberFunctions`?

Reply:
@import-brain Can you help us with the wording here?

Reply:
> @import-brain Can you help us with the wording here?

Because we're binding `this` to the member functions, `bindThisToMemberFunctions` would be the best wording in this case.